### ISSUE TITLE
Support node 6?

### ISSUE THREAD

**Issue by hems** [allenluce/mmap-object#14]
Unfortunately i have some modules which requires node 6, but the latest version i managed to install mmap was 5.

How hard would be to update it to be compatible with 6?

Thank You

**Comment by dougluce** [comment]
It should have no troubles with Node versions from 4 through 9.  [Travis builds these](https://travis-ci.org/allenluce/mmap-object) without much trouble.

What happened when you tried to use the module with Node 6? Post output if you would please.

**Comment by hems** [comment]
I had a look at Travis builds and it looks good indeed, but locally, this is what happens when i clone the repo and use  ````--build-from-source````, am i missing some dependency? looks lik it depends on boost?

````bash
replay/mmap-object [ node --version                                                                                                                                                                                              master ] 7:37 PM
v8.2.1
replay/mmap-object [ npm --version                                                                                                                                                                                               master ] 7:37 PM
5.3.0
replay/mmap-object [ npm install --build-from-source                                                                                                                                                                             master ] 7:37 PM

> mmap-object@1.3.7 install /Volumes/backpack/git/hemz/sozo/replay/mmap-object
> node-pre-gyp install --fallback-to-build

gyp WARN download NVM_NODEJS_ORG_MIRROR is deprecated and will be removed in node-gyp v4, please use NODEJS_ORG_MIRROR
gyp WARN download NVM_NODEJS_ORG_MIRROR is deprecated and will be removed in node-gyp v4, please use NODEJS_ORG_MIRROR
gyp WARN download NVM_NODEJS_ORG_MIRROR is deprecated and will be removed in node-gyp v4, please use NODEJS_ORG_MIRROR
  CXX(target) Release/obj.target/mmap-object/mmap-object.o
../mmap-object.cc:11:10: fatal error: 'boost/interprocess/managed_mapped_file.hpp' file not found
#include <boost/interprocess/managed_mapped_file.hpp>
         ^
1 error generated.
make: *** [Release/obj.target/mmap-object/mmap-object.o] Error 1
gyp ERR! build error
gyp ERR! stack Error: `make` failed with exit code: 2
gyp ERR! stack     at ChildProcess.onExit (/Users/h/.nvm/versions/node/v8.2.1/lib/node_modules/npm/node_modules/node-gyp/lib/build.js:258:23)
gyp ERR! stack     at emitTwo (events.js:125:13)
gyp ERR! stack     at ChildProcess.emit (events.js:213:7)
gyp ERR! stack     at Process.ChildProcess._handle.onexit (internal/child_process.js:200:12)
gyp ERR! System Darwin 16.0.0
gyp ERR! command "/Users/h/.nvm/versions/node/v8.2.1/bin/node" "/Users/h/.nvm/versions/node/v8.2.1/lib/node_modules/npm/node_modules/node-gyp/bin/node-gyp.js" "build" "--fallback-to-build" "--module=/Volumes/backpack/git/hemz/sozo/replay/mmap-object/lib/mmap-object.node" "--module_name=mmap-object" "--module_path=/Volumes/backpack/git/hemz/sozo/replay/mmap-object/lib" "--napi_version=1" "--node_abi_napi=napi"
gyp ERR! cwd /Volumes/backpack/git/hemz/sozo/replay/mmap-object
gyp ERR! node -v v8.2.1
gyp ERR! node-gyp -v v3.6.2
gyp ERR! not ok
node-pre-gyp ERR! build error
node-pre-gyp ERR! stack Error: Failed to execute '/Users/h/.nvm/versions/node/v8.2.1/bin/node /Users/h/.nvm/versions/node/v8.2.1/lib/node_modules/npm/node_modules/node-gyp/bin/node-gyp.js build --fallback-to-build --module=/Volumes/backpack/git/hemz/sozo/replay/mmap-object/lib/mmap-object.node --module_name=mmap-object --module_path=/Volumes/backpack/git/hemz/sozo/replay/mmap-object/lib --napi_version=1 --node_abi_napi=napi' (1)
node-pre-gyp ERR! stack     at ChildProcess.<anonymous> (/Volumes/backpack/git/hemz/sozo/replay/mmap-object/node_modules/node-pre-gyp/lib/util/compile.js:83:29)
node-pre-gyp ERR! stack     at emitTwo (events.js:125:13)
node-pre-gyp ERR! stack     at ChildProcess.emit (events.js:213:7)
node-pre-gyp ERR! stack     at maybeClose (internal/child_process.js:921:16)
node-pre-gyp ERR! stack     at Process.ChildProcess._handle.onexit (internal/child_process.js:211:5)
node-pre-gyp ERR! System Darwin 16.0.0
node-pre-gyp ERR! command "/Users/h/.nvm/versions/node/v8.2.1/bin/node" "/Volumes/backpack/git/hemz/sozo/replay/mmap-object/node_modules/.bin/node-pre-gyp" "install" "--fallback-to-build"
node-pre-gyp ERR! cwd /Volumes/backpack/git/hemz/sozo/replay/mmap-object
node-pre-gyp ERR! node -v v8.2.1
node-pre-gyp ERR! node-pre-gyp -v v0.9.0
node-pre-gyp ERR! not ok
Failed to execute '/Users/h/.nvm/versions/node/v8.2.1/bin/node /Users/h/.nvm/versions/node/v8.2.1/lib/node_modules/npm/node_modules/node-gyp/bin/node-gyp.js build --fallback-to-build --module=/Volumes/backpack/git/hemz/sozo/replay/mmap-object/lib/mmap-object.node --module_name=mmap-object --module_path=/Volumes/backpack/git/hemz/sozo/replay/mmap-object/lib --napi_version=1 --node_abi_napi=napi' (1)
npm ERR! code ELIFECYCLE
npm ERR! errno 1
npm ERR! mmap-object@1.3.7 install: `node-pre-gyp install --fallback-to-build`
npm ERR! Exit status 1
npm ERR!
npm ERR! Failed at the mmap-object@1.3.7 install script.
npm ERR! This is probably not a problem with npm. There is likely additional logging output above.

npm ERR! A complete log of this run can be found in:
npm ERR!     /Users/h/.npm/_logs/2018-03-28T18_37_49_015Z-debug.log
````

**Comment by hems** [comment]
after "brew install boost" it worked, it's my fault i didn't realise "boost" was a requirement

**Comment by allenluce** [comment]
Glad you got it going.

Looks like a [regression in `node-pre-gyp`](https://github.com/mapbox/node-pre-gyp/issues/359) contributed to your original problem.  I've [submitted a PR](https://github.com/mapbox/node-pre-gyp/pull/361) to fix `node-pre-gyp`. I'm re-opening this issue so others can find it. Once `node-pre-gyp` is fixed I'll close it.

**Comment by allenluce** [comment]
The PR was merged and 0.9.1/0.10.0 have been released. I'm closing this again, please let me know if you continue to have problems with installation.
